src/main/java/com/example/onlinebookstoremy/bookstore/dto/request/BookSearchParametersDto.java

```diff
@@ -0,0 +1,4 @@
+package com.example.onlinebookstoremy.bookstore.dto.request;
+
+public record BookSearchParametersDto(String[] titles, String[] authors) {
```

Comment:
As I see you want to realize CriteriaAPI in your project, but where can I find the realization? ;)